app/controllers/api/providers_controller.rb

```diff
@@ -228,6 +233,12 @@ def create_provider_ddf(data)
       raise BadRequestError, "Could not create the new provider - #{err}"
     end
 
+    def edit_provider_ddf(provider, data)
+      provider.edit_with_params(data)
```

Comment:
does this also do what edit_provider does w.r.t. updating provider authentication and such ?

Reply:
Yes, but differently. Each provider implements it on its own as the incoming data is in a different (DDF) format.